app/src/main/java/com/dragic/gamehunter/viewmodel/GameDetailsViewModel.kt

```diff
@@ -0,0 +1,67 @@
+package com.dragic.gamehunter.viewmodel
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.dragic.gamehunter.di.BG_DISPATCHER
+import com.dragic.gamehunter.di.GameId
+import com.dragic.gamehunter.repository.DealRepository
+import com.dragic.gamehunter.utils.toDealDetailsViewState
+import com.dragic.gamehunter.utils.toImageContentViewState
+import com.dragic.gamehunter.view.gamedetails.DealDetailsViewState
+import com.dragic.gamehunter.view.gamedetails.ImageContentViewState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import javax.inject.Named
+import kotlin.coroutines.CoroutineContext
+
+const val DEAL_REDIRECT_LINK_PREFIX = "https://www.cheapshark.com/redirect?dealID="
+
+@HiltViewModel
+class GameDetailsViewModel @Inject constructor(
+    private val repository: DealRepository,
+    @GameId private val gameId: Int,
+    @Named(BG_DISPATCHER) private val bgDispatcher: CoroutineContext,
+) : ViewModel() {
+    var gameData by mutableStateOf<ImageContentViewState?>(null)
+        private set
+    var dealData by mutableStateOf<List<DealDetailsViewState>>(emptyList())
+        private set
+
+    init {
+        viewModelScope.launch(bgDispatcher) {
+            repository.fetchGameDetailsData(gameId)
+        }
+        viewModelScope.launch(bgDispatcher) {
+            repository.fetchStoreInfo()
+        }
+        viewModelScope.launch(bgDispatcher) {
+            repository.gameDetailsData(gameId).collect { gameDetails ->
+                gameData = gameDetails.toImageContentViewState()
+            }
+        }
+        viewModelScope.launch(bgDispatcher) {
+            repository.getGameDealsDetails().collect { dealDetails ->
+                dealData = dealDetails.map { it.toDealDetailsViewState() }
+            }
+        }
```

Comment:
Coroutines should make sure they're not blocking the main thread on their own. The ViewModel should never schedule Repository and other domain-layer methods on the appropriate dispatchers. This should be handled internally by the Repository. The ViewModel then only needs to call `viewModelScope.launch {...}` 

There are more occurrences of this, so please address them too